### ISSUE TITLE
0.19.0 gives error when building with Carthage

### ISSUE THREAD

**Issue by entotsu** [SnapKit/SnapKit#190]
0.18.0 -> 0.19.0
>  carthage update --platform iOS  

** BUILD FAILED **

**Comment by robertjpayne** [comment]
@entotsu can you provide any other information? What version of Carthage are you using? What iOS version are you targeting? Any other build output?

It's very hard to debug what is happening here without seeing more info.

On my machine Carthage builds just fine.

**Comment by Vaseltior** [comment]
Hi, here is perhaps the kind of information you're looking for :
```
The following build commands failed:
	CodeSign /Users/xxx/Library/Developer/Xcode/DerivedData/SnapKit-ezapyuzzsoeigdgyxhdpufyxokys/Build/Products/Release-iphoneos/SnapKit.framework
	CodeSign /Users/xxx/Library/Developer/Xcode/DerivedData/SnapKit-ezapyuzzsoeigdgyxhdpufyxokys/Build/Products/Release-iphoneos/SnapKit.framework
```

Yesterday, there was no problem compiling `SnapKit`. I guess you have made a substantial modification to your project that could issue this.

**Comment by Vaseltior** [comment]
this commit `245a5fcf6e1a7836d984b49a432f99e47f1c5aed ` compiles correctly, and then, after that, it doesn't.

**Comment by robertjpayne** [comment]
@Vaseltior @entotsu What are your Cartfiles like?

If I create a brand new folder with a Cartfile of:
```
github "snapkit/snapkit" >= 0.19.1
```

And then run `carthage bootstrap` it builds all three frameworks just fine with no errors. 

CodeSigning sounds like you may not have any distribution identities available on your machine?

**Comment by saiGo1992** [comment]
How to use your framework to suit different size of screen? like iPhone4s,iPhone5, iPhone6, all of them have differentsize?  can make your framework auto suit different size of screen

**Comment by Vaseltior** [comment]
I'm getting the develop branch, which is risky, indeed.
Here is what my Cartfile look like around SnapKit. I'll give a try to get the tagged version of SnapKit, thanks for that @robertjpayne . But I think @entotsu may have another problem.

```
...
# SnapKit
github "SnapKit/SnapKit" "245a5fcf6e1a7836d984b49a432f99e47f1c5aed"
# PromiseKit
github "mxcl/PromiseKit" "swift-2.0-beta5"
# Alamofire
github "Alamofire/Alamofire" >= 1.2
...
```

**Comment by Vaseltior** [comment]
Ok, Please have a look at this :

```
...
# SnapKit
github "SnapKit/SnapKit">= 0.19.1
...
```

Gives us the following result...

```
The following build commands failed:
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
	CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler
	Ld /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit\ tvOS.build/Objects-normal/armv7/SnapKit normal armv7
	Ld /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit\ tvOS.build/Objects-normal/arm64/SnapKit normal arm64
	CreateUniversalBinary /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit.framework/SnapKit normal armv7\ arm64
	GenerateDSYMFile /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit.framework.dSYM /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit.framework/SnapKit
	Ld /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit\ iOS\ Tests.build/Objects-normal/armv7/SnapKit\ iOS\ Tests normal armv7
	Ld /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit\ iOS\ Tests.build/Objects-normal/arm64/SnapKit\ iOS\ Tests normal arm64
	CreateUniversalBinary /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit\ iOS\ Tests.xctest/SnapKit\ iOS\ Tests normal armv7\ arm64
	GenerateDSYMFile /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit\ iOS\ Tests.xctest.dSYM /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit\ iOS\ Tests.xctest/SnapKit\ iOS\ Tests
(10 failures)
/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SFCore/SFCoreTests/String+SFCharactersTests.swift:55: warning: Line Length Violation: Line should be 120 characters or less: currently 128 characters (line_length)
/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SFCore/SFCoreTests/String+SFCharactersTests.swift:57: warning: Line Length Violation: Line should be 120 characters or less: currently 137 characters (line_length)
/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SFCore/SFCoreTests/String+SFCharactersTests.swift:55: warning: Line Length Violation: Line should be 120 characters or less: currently 128 characters (line_length)
/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SFCore/SFCoreTests/String+SFCharactersTests.swift:57: warning: Line Length Violation: Line should be 120 characters or less: currently 137 characters (line_length)
<unknown>:0: error: no such file or directory: '/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SnapKit/SourceLocation.swift'
<unknown>:0: error: no such file or directory: '/Users/S000000/Documents/projects/aviva-expertise-grele/Expertise-Grele/Carthage/Checkouts/SnapKit/SourceLocation.swift'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit tvOS.build/Objects-normal/armv7/SnapKit (No such file or directory)
error: cannot parse the debug map for "/Users/S077128/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit.framework/SnapKit": No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/S000000/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Intermediates/SnapKit.build/Release-iphoneos/SnapKit iOS Tests.build/Objects-normal/armv7/SnapKit iOS Tests (No such file or directory)
error: cannot parse the debug map for "/Users/S077128/Library/Developer/Xcode/DerivedData/SnapKit-auggxvpypbuykdbtfbxwlnygqfjy/Build/Products/Release-iphoneos/SnapKit iOS Tests.xctest/SnapKit iOS Tests": No such file or directory
A shell task failed with exit code 65:
** BUILD FAILED **
```

**Comment by robertjpayne** [comment]
@Vaseltior develop should work but you definitely risk getting a copy that has build errors. Closing this issue as it seems to be configuration related rather than SnapKit's issue.

Please ensure you are using the latest tagged release for Carthage.

**Comment by robertjpayne** [comment]
@Vaseltior also GitHub bugging up a bit just saw email with the full build log info and stuff ( maybe you deleted that comment ).

I think Carthage may have cached a checkout of SnapKit that isn't the right version. That was an issue I resolved further down the road before 0.19.1 release.

**Comment by entotsu** [comment]
thanks 
nice 
good
god
